src/custom/state/lists/hooks/hooksMod.ts

```diff
@@ -55,40 +55,41 @@ import { isAddress } from 'utils'
 //   [ChainId.XDAI]: {}
 // }
 
-// const listCache: WeakMap<TokenList, TokenAddressMap> | null =
-//   typeof WeakMap !== 'undefined' ? new WeakMap<TokenList, TokenAddressMap>() : null
-
-// export function listToTokenMap(list: TokenList): TokenAddressMap {
-//   const result = listCache?.get(list)
-//   if (result) return result
-
-//   const map = list.tokens.reduce<TokenAddressMap>(
-//     (tokenMap, tokenInfo) => {
-//       const tags: TagInfo[] =
-//         tokenInfo.tags
-//           ?.map(tagId => {
-//             if (!list.tags?.[tagId]) return undefined
-//             return { ...list.tags[tagId], id: tagId }
-//           })
-//           ?.filter((x): x is TagInfo => Boolean(x)) ?? []
-//       const token = new WrappedTokenInfo(tokenInfo, tags)
-//       if (tokenMap[token.chainId][token.address] !== undefined) throw Error('Duplicate tokens.')
-//       return {
-//         ...tokenMap,
-//         [token.chainId]: {
-//           ...tokenMap[token.chainId],
-//           [token.address]: {
-//             token,
-//             list: list
-//           }
-//         }
-//       }
-//     },
-//     { ...EMPTY_LIST }
-//   )
-//   listCache?.set(list, map)
-//   return map
-// }
+const listCache: WeakMap<TokenList, TokenAddressMap> | null =
+  typeof WeakMap !== 'undefined' ? new WeakMap<TokenList, TokenAddressMap>() : null
+
+export function listToTokenMap(list: TokenList): TokenAddressMap {
+  const result = listCache?.get(list)
+  if (result) return result
+
+  const map = list.tokens.reduce<TokenAddressMap>(
+    (tokenMap, tokenInfo) => {
+      const tags: TagInfo[] =
+        tokenInfo.tags
+          ?.map(tagId => {
+            if (!list.tags?.[tagId]) return undefined
+            return { ...list.tags[tagId], id: tagId }
+          })
+          ?.filter((x): x is TagInfo => Boolean(x)) ?? []
+      const token = new WrappedTokenInfo(tokenInfo, tags)
+      const tokensByNetwork = tokenMap[token.chainId] || {}
```

Comment:
Uncommenting original "Mod" code, adding this extra line (75)

Reply:
i think we should just do the change i made for uni when it broke the first time (it's related right?)
https://github.com/Uniswap/uniswap-interface/issues/1383

Reply:
ah it's a bit different, anyways nice.

Reply:
yep, just fixing the issue with "unknown" networks. 
Your change also makes sense, but I would add it in develop since a hotfix always tries to be as minimal as possible to fix the issue